src/main/kotlin/HttpEndpoint.kt

```diff
@@ -811,6 +826,35 @@ class HttpEndpoint : CoroutineVerticle() {
     renderAsset("ui/new/workflow/index.html", ctx.response())
   }
 
+  /**
+   * Check if Steep is healthy.
+   * @param ctx the routing context
+   */
+  private fun onGetHealth(ctx: RoutingContext) = launch {
+    try {
+      if (!ready) {
+        throw RuntimeException("Steep is not ready yet")
+      }
+      val result = json {
+        obj(
+          "services" to metadataRegistry.findServices().size,
+          "agents" to agentRegistry.getAgentIds().size,
+          "submissions" to submissionRegistry.countSubmissions(),
+          "vms" to vmRegistry.countVMs(),
+          "health" to true
+        )
+      }
+      ctx.response()
+        .putHeader("content-type", "application/json")
+        .end(result.encode())
+    } catch (t: Throwable) {
+      ctx.response()
```

Comment:
It would be perfect if the application could report why it's not healthy, i.e. if it could render a full JSON object like above, where only those components that are not ready are `false`. This requires a separate try-catch block around each of the calls to the registries.